processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -407,35 +216,56 @@ public T get(int index)
   @Override
   public int indexOf(T value)
   {
-    return bufferIndexed.indexOf(value);
+    return indexOf(this, value);
+  }
+
+  private int indexOf(Indexed<T> indexed, T value)
```

Comment:
`indexOf()` is not dramatically slower. It's just the same (before-after):
```java
GenericIndexedBenchmark.get                  8  50000  avgt   20   46.342 ±  2.523  ns/op
GenericIndexedBenchmark.indexOf              8  50000  avgt   20  663.044 ± 10.115  ns/op

GenericIndexedBenchmark.get                  8  50000  avgt   20   45.611 ± 0.584  ns/op
GenericIndexedBenchmark.indexOf              8  50000  avgt   20  664.070 ± 7.671  ns/op
```

Also `indexOf()` is not a hotspot, unlike `get()`.

Benchmark results with v2 are highly unstable on my machine (because of the data size, disk IO, etc.) Why do you expect any surprises here? v2 is changed similarly to v1.

Didn't compare with pre-v2, because it's not actionable. If it is slightly faster than newer versions, what then? Revert the v2 PR?